admin/src/pages/teams/connected-teams-list.tsx

```diff
@@ -0,0 +1,311 @@
+	disconnectTeam,
+	FigmaTeamAuthStatus,
+	type FigmaTeamSummary,
+} from '../../api';
+import {
+	FigmaPermissionsPopup,
+	Link,
+	Page,
+	SuccessBanner,
+} from '../../components';
+import { pluralize } from '../../utils';
+
+type ConnectedTeamsListProps = {
+	teams: readonly FigmaTeamSummary[];
+	addTeam: () => void;
+	site: string;
+};
+
+export function ConnectedTeamsList({
+	teams,
+	addTeam,
+	site,
+}: ConnectedTeamsListProps) {
+	const queryClient = useQueryClient();
+	const disconnectMutation = useMutation({
+		mutationFn: (teamId: string) => {
+			return disconnectTeam(teamId);
+		},
+		onSuccess: () => {
+			return queryClient.invalidateQueries({ queryKey: ['teams'] });
+		},
+	});
+
+	const reconnectMutation = useMutation({
+		mutationFn: (teamId: string) => {
+			return connectTeam(teamId);
+		},
+		onSuccess: () => {
+			return queryClient.invalidateQueries({ queryKey: ['teams'] });
+		},
+	});
+
+	return (
+		<Page>
+			<div
+				css={css`
+					display: flex;
+					flex-direction: column;
+					align-items: center;
+					gap: 32px;
+					width: 580px;
+				`}
+			>
+				<div
+					css={css`
+						display: flex;
+						flex-direction: column;
+						align-items: center;
+						gap: 8px;
+					`}
+				>
+					<SuccessBanner size="small" />
+					<div
+						css={css`
+							color: ${token('color.text')};
+							font-size: 24px;
+							font-weight: 500;
+						`}
+					>
+						Connected Figma teams
+					</div>
+				</div>
+				<div
+					css={css`
+						color: ${token('color.text.subtle')};
+					`}
+				>
+					You have {teams.length} Figma{' '}
+					{pluralize(teams.length, 'team', 'teams')} connected to{' '}
+					<strong>{site}</strong>
+				</div>
+				{disconnectMutation.isError && (
+					<DisconnectTeamUnauthorizedSectionMessage />
+				)}
+				{reconnectMutation.isError && <GenericErrorSectionMessage />}
+				<div
+					css={css({
+						padding: '32px 40px',
+						borderWidth: 2,
+						borderColor: token('color.border'),
+						borderStyle: 'solid',
+						borderRadius: 3,
+						display: 'flex',
+						flexDirection: 'column',
+						alignSelf: 'stretch',
+					})}
+				>
+					{teams.map((team) => (
+						<ConnectedTeam
+							key={team.teamId}
+							team={team}
+							disconnectTeam={disconnectMutation.mutate}
+							reconnectTeam={reconnectMutation.mutate}
+						/>
+					))}
+					<div
+						css={css`
+							display: flex;
+							padding: 12px 16px;
+							align-items: center;
+							gap: 12px;
+						`}
+					>
+						<Button
+							onClick={addTeam}
+							iconBefore={<AddIcon size="medium" label="" />}
+						></Button>
+						<span
+							css={css`
+								flex-grow: 1;
+								font-weight: 500;
+							`}
+						>
+							Add another team
+						</span>
+					</div>
+				</div>
+			</div>
+		</Page>
+	);
+}
+
+type ConnectedTeamProps = {
+	team: FigmaTeamSummary;
+	disconnectTeam: (teamId: string) => void;
+	reconnectTeam: (teamId: string) => void;
+};
+
+function ConnectedTeam({
```

Comment:
nit: this file is pretty chonky, thoughts on pulling this (and potentially the error components) out into their own files?